frontend/catalyst/utils/jax_extras.py

```diff
@@ -636,3 +636,16 @@ def _gather_shape_rule_dynamic(
             )
 
     return _gather_shape_computation(indices, dimension_numbers, slice_sizes)
+
+
+def remove_host_context(jaxpr):
+    """This function will remove the host context.
```

Comment:
What do you mean by host context?

Reply:
The host context is the wrapper function that calls a qnode. When applying a `@qjit` decorator immediately on a qnode like:

```
@qjit
@qml.qnode(qml.device("lightning.qubit", wires=1))
def identity(x):
  return x
```

Notice that in the JAXPR (and all subsequent IRs), we have this wrapper context that only performs a function call.

```
{ lambda ; a:i64[]. let
    b:i64[] = func[
      call_jaxpr={ lambda ; c:i64[]. let
           = qdevice[
            rtd_kwargs={'shots': 0, 'mcmc': False}
            rtd_lib=/home/ubuntu/code/catalyst/frontend/catalyst/utils/../../../runtime/build/lib/librtd_lightning.so
            rtd_name=LightningSimulator
          ] 
          d:AbstractQreg() = qalloc 1
           = qdealloc d
        in (c,) }
      fn=<QNode: wires=1, device='lightning.qubit', interface='auto', diff_method='best'>
    ] a
  in (b,) }
```

Reply:
Oh got it 👍  What would happen if I applied `qjit` on something other than a QNode?

Reply:
It will now raise a CompileError.